awx/main/utils/filters.py

```diff
@@ -110,6 +110,17 @@ def filter(self, record):
             return bool(base_name in loggers)
 
 
+class GlobalLogLevelFilter(Filter):
+    """Configure-tower-in-tower setting LOG_AGGREGATOR_LEVEL
+    shall be used to control the log levels of all logs in the system
+    including those writing to file
+    """
+
+    def filter(self, record):
+        allowed_level = settings.LOG_AGGREGATOR_LEVEL
```

Comment:
If the database is down/unavailable, what happens here?

Reply:
we have logic in conf/settings where we return the default.

However... I think you still might be on to something, and we need to tread very carefully here.

In the other filter class, I was maintaining a logger blacklist. Many of those blacklist items are not applicable to this type of filter. If you have an error sending a log to an external collector, then sure, log the error sending the log to file.

However, if we fail to connect to the database... we get traceback details... and we log that...